File: Java/src/Stock.java
import java.util.Map;
import java.util.HashMap;

public class Stock {

    public String[] findHighestAverageStocks(String[] stocks, double[][] prices) {
        // Création d'une map pour stocker les moyennes par action
        Map<String, Double> averages = new HashMap<>();

        // Calcul de la moyenne pour chaque action
        for (int i = 0; i < stocks.length; i++) {
            double sum = 0;
            for (double price : prices[i]) {
                sum += price;
            }
            double average = sum / prices[i].length;
            averages.put(stocks[i], average);
        }

        // Tri des actions par moyenne décroissante
        return averages.entrySet()
                .stream()
                .sorted(Map.Entry.<String, Double>comparingByValue().reversed())
                .limit(3)
                .map(Map.Entry::getKey)
                .toArray(String[]::new);
    }
}

File: TS/stock.ts
// Retourne les 3 actions ayant la moyenne de prix la plus elevee (ordre decroissant)
function findHighestAverageStocks(stocks: string[], prices: number[][]): string[] {
  const averages: { stock: string; average: number }[] = [];

  for (let i = 0; i < stocks.length; i++) {
    const sum = prices[i].reduce((acc, price) => acc + price, 0);
    averages.push({ stock: stocks[i], average: sum / prices[i].length });
  }

  return averages
    .sort((a, b) => b.average - a.average)
    .slice(0, 3)
    .map(entry => entry.stock);
}

export { findHighestAverageStocks };
